cli/cli/commands/run/run.go

```diff
@@ -734,3 +824,92 @@ func isHttpUrl(maybeHttpUrl string) bool {
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred retrieving Docker Kurtosis Backend")
+	}
+	for _, img := range images {
+		_, _, err := kurtosisBackend.FetchImage(ctx, img, nil, image_download_mode.ImageDownloadMode_Always)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred pulling '%v' locally.", img)
+		}
+	}
+	return nil
+}
+
+func pullPackagesLocally(packageDependencies []string) (map[string]string, error) {
+	localPackagesToRelativeFilepaths := map[string]string{}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting current working directory.")
+	}
+	parentCwd := filepath.Dir(wd)
+	relParentCwd, err := filepath.Rel(wd, parentCwd)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting rel path between '%v' and '%v'.", wd, relParentCwd)
+	}
+
+	for _, dependency := range packageDependencies {
+		packageIdParts := strings.Split(dependency, "/")
+		packageName := packageIdParts[len(packageIdParts)-1]
+		logrus.Infof("Pulling package: %v", dependency)
+
+		// assume dependency doesn't have http:// prefix or .git suffix
+		repoUrl := fmt.Sprintf("%s%s%s", "http://", dependency, ".git")
+		localPackagePath := fmt.Sprintf("%s/%s", parentCwd, packageName)
+		logrus.Infof("repo url: %v", repoUrl)
+		_, err := git.PlainClone(localPackagePath, false, &git.CloneOptions{
+			URL:               repoUrl,
+			Auth:              nil,
+			RemoteName:        "",
+			ReferenceName:     "",
+			SingleBranch:      false,
+			Mirror:            false,
+			NoCheckout:        false,
+			Depth:             0,
+			RecurseSubmodules: 0,
+			ShallowSubmodules: false,
+			Progress:          nil,
+			Tags:              0,
+			InsecureSkipTLS:   false,
+			CABundle:          nil,
+			ProxyOptions: transport.ProxyOptions{
+				URL:      "",
+				Username: "",
+				Password: "",
+			},
+			Shared: false,
+		})
+		if err != nil && !errors.Is(err, git.ErrRepositoryAlreadyExists) {
+			return nil, stacktrace.Propagate(err, "An error occurred cloning package '%s' to '%s'.", dependency, localPackagePath)
+		}
+		localPackagesToRelativeFilepaths[dependency] = fmt.Sprintf("%s/%s", relParentCwd, packageName)
+	}
+
+	return localPackagesToRelativeFilepaths, nil
+}
+
+func updateKurtosisYamlWithReplaceDirectives(packageNamesToLocalFilepaths map[string]string) error {
+	file, err := os.OpenFile(kurtosisYMLFilePath, os.O_WRONLY|os.O_APPEND, kurtosisYMLFilePerms)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred opening kurtosis.yml file.")
+	}
+	defer file.Close()
+
+	_, err = file.Write([]byte("replace:\n"))
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred writing 'replace:' to kurtosis.yml")
+	}
+
+	for packageName, localFilepath := range packageNamesToLocalFilepaths {
+		_, err := file.Write([]byte(fmt.Sprintf("  %s: %s\n", packageName, localFilepath)))
```

Comment:
Potential bug: we're assuming the user's YAML is indented by two spaces per indentation level; seems like it could also be 4 spaces, tabs, etc. If it's hard to fix in a proper way, maybe we could at least add a TODO here showing that we recognize the risk and how to address this properly?